explorer/testdata/namespace/lookup_scope.carbon

```diff
@@ -0,0 +1,21 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// RUN: %{explorer-run}
+// RUN: %{explorer-run-trace}
+// CHECK:STDOUT: result: 0
+
+package ExplorerTest api;
+
+namespace N;
```

Comment:
This isn't really testable yet, because qualified namespace declarations aren't supported. We'd reject the second line of your example. #2575 adds support for qualified namespaces, and has some minimal testing for this, but testing the location of the error sounds like a nice change. I'll add that test there.